### ISSUE TITLE
Invalid behavior for `ops.take` with torch

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras#18449]
It appears that `ops.take(embed_matrix, ids)` has a different behavior in torch compared to the other 2 backends.

**Comment by sachinprasadhs** [comment]
I tried the below code with Torch, it is now giving same result as TensorFlow and Jax.

```python
import os
os.environ["KERAS_BACKEND"] = "torch"
import numpy as np
import keras

a = np.array([[0, 1.0, 2.0],
              [10.0, 11.0, 12.0],
              [20.0, 21.0, 22.0],
              [30.0, 31.0, 32.0]])
print(keras.ops.take(a,[3,1],axis=0))

tensor([[30., 31., 32.],
        [10., 11., 12.]], dtype=torch.float64)
```

Only in out of range index, torch throws `IndexError`, `JAX` handles with `nan` values and `TensorFlow` provides the last axis values.

**Comment by innat-asj** [comment]
FYI, https://github.com/keras-team/keras/issues/19238#issuecomment-2008046857
